### PR TITLE
Remove authentication from group search

### PR DESCRIPTION
**Pull request by trezy**
So many more changes.

* Remove authentication from the groups page
* Remove `nodemon` from the Docker container because it sometimes loses track of files being changes
* Update a lot of Dialog styles
* Create a Registration Dialog that's shown when the user clicks on the "Join Group" button but isn't logged in
* Add better styles for disabled inputs and input groups
* Tall dialogs now cause the page to be scrollable, rather than just the Dialog's content

### CODE REVIEW

File: pages/groups/group.js [hunk 1]
```diff
@@ -188,6 +187,10 @@ class GroupProfile extends Component {
     window.location.reload()
   }
 
+  _hideRegistrationModal () {
```
Comment (UncleClapton) [inline pages/groups/group.js]: I think having dedicated functions for setting a single state value like this might introduce a lot of clutter very quickly. keep in mind that I also have another PR coming up which adds another dialog to the groups page. Thoughts on a solution?

File: pages/groups/group.js [hunk 2]
```diff
@@ -561,6 +570,12 @@ class GroupProfile extends Component {
             </TabPanel>
           </div>
         </Main>
+
+        {Boolean(showRegistrationModal) && (
```
Comment (UncleClapton) [inline pages/groups/group.js]: The `Boolean` cast here is superfluous here since we only ever expect `showRegistrationModal` to be `true` or `false`

File: pages/groups/group.js (hunk 1 above)
Comment (trezy) [reply pages/groups/group.js]: I'm good with sticking to using `setState` directly instead of independent functions for it. I'll update that.

As for having multiple modals, it should be fine as long as they have independent display states, right?

File: pages/groups/group.js (hunk 2 above)
Comment (trezy) [reply pages/groups/group.js]: Legit. I'll update.

File: pages/groups/group.js (hunk 1 above)
Comment (UncleClapton) [reply pages/groups/group.js]: yeah I'm not worried about conflicting state values, I'm more worried about 30 lines of `(show/hide)Modal` functions for no good reason :P

tl;dr: I hate Java
